workflow/state-machines.rst

```diff
@@ -0,0 +1,197 @@
+.. index::
+    single: Workflow; Workflows as State Machines
+
+Workflows as State Machines
+===========================
+
+The workflow component is modelled after a *Workflow net* which is a subclass
+of a `Petri net`_. By adding further restrictions you can get a state machine.
+The most important one being that a state machine cannot be in more than
+one place simultaneously. It is also worth noting that a workflow does not
+commonly have cyclic path in the definition graph, but it is common for a state
+machine.
+
+Example of a State Machine
+--------------------------
+
+A pull request starts in an intial "start" state, a state for e.g. running
+tests on Travis. When this is finished, the pull request is in the "review"
+state, where contributors can require changes, reject or accept the
+pull request. At any time, you can also "update" the pull request, which
+will result in another Travis run.
+
+.. image:: /_images/components/workflow/pull_request.png
+
+Below is the configuration for the pull request state machine.
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # app/config/config.yml
+        framework:
+            workflows:
+                pull_request:
+                   type: 'state_machine'
+                   supports:
+                        - AppBundle\Entity\PullRequest
+                   places:
+                        - start
+                        - coding
+                        - travis
+                        - review
+                        - merged
+                        - closed
+                   transitions:
+                        submit:
+                            from: start
+                            to: travis
+                        update:
+                            from: [coding, travis, review]
+                            to: travis
+                        wait_for_reivew:
+                            from: travis
+                            to: review
+                        change_needed:
+                            from: review
+                            to: coding
+                        accepted:
+                            from: review
+                            to: merged
+                        rejected:
+                            from: review
+                            to: closed
+                        reopened:
```

Comment:
What do you think about renaming the last four transitions to `request_change`, `accept`, `reject`, and `reopen` as we are describing actions and not events that just happened?